### ISSUE TITLE
need version bump, otherwise folks will have a bad time

### ISSUE THREAD

**Issue by jzacsh** [etesync/etebase-rs#25]
**tl;dr** just push a git tag to represent the fix of issue #22 like commit `bce20af56b55399` (eg `0.6.0` or `0.5.1` or something)

## justification

If you don't push a tag to indicate the latest version is fixed, crates.io still shows `0.5.0` as the latest version to add into your dependencies (as a rust user), in the side-bar of crates.io's UI. Then you'll follow that instruction, find an error, and then start spending time trying to file a helpful bug before realizing there's already mention of this exact compilation failure in issue #22 :P

<details>
<summary>steps to have a bad time, as a new user of this lib</summary>

### step 1: fresh crate initialization via `cargo new --bin`

```sh
$ cargo --version
cargo 1.57.0 (b2e52d7ca 2021-10-21)
$ cargo new --bin foo
$ cd foo
```

### step 2: add dependency on `etebase` crate

`$EDITOR Cargo.toml` and add etebase crate dependency:
```diff
diff --git a/Cargo.toml b/Cargo.toml
index c29b9bb..6fcf505 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,3 +6,4 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

 [dependencies]
+etebase = "0.5.0"
```
### step 3: `cargo build` without touching anything else

`cargo build` will now lead to 4 rust compilation errors in etebase...

```sh
$ cargo build
    Updating crates.io index
  Downloaded form_urlencoded v1.0.1

# ... snipped out for brevity...

   Compiling etebase v0.5.0
error[E0423]: cannot initialize a tuple struct which contains private fields
   --> /home/jzacsh/usr/lib/rust/cargo/registry/src/github.com-1ecc6299db9ec823/etebase-0.5.0/src/crypto.rs:193:25
    |
193 |         let signature = sign::Signature(signature_copy);
    |                         ^^^^^^^^^^^^^^^
    |
note: constructor is not visible here due to private fields
   --> /home/jzacsh/usr/lib/rust/cargo/registry/src/github.com-1ecc6299db9ec823/ed25519-1.3.0/src/lib.rs:305:22
    |
305 | pub struct Signature([u8; Signature::BYTE_SIZE]);
    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^ private field

error[E0308]: mismatched types
  --> /home/jzacsh/usr/lib/rust/cargo/registry/src/github.com-1ecc6299db9ec823/etebase-0.5.0/src/crypto.rs:28:59
   |
28 |     let mut state = to_enc_error!(generichash::State::new(32, key), "Failed to init hash")?;
   |                                                           ^^
   |                                                           |
   |                                                           expected enum `std::option::Option`, found integer
   |                                                           help: try using a variant of the expected enum: `online_managers::_::_serde::__private::Some(32)`
   |
   = note: expected enum `std::option::Option<usize>`
              found type `{integer}`

error[E0608]: cannot index into a value of type `Signature`
   --> /home/jzacsh/usr/lib/rust/cargo/registry/src/github.com-1ecc6299db9ec823/etebase-0.5.0/src/crypto.rs:187:12
    |
187 |         Ok(ret[..].to_vec())
    |            ^^^^^^^

error[E0308]: mismatched types
   --> /home/jzacsh/usr/lib/rust/cargo/registry/src/github.com-1ecc6299db9ec823/etebase-0.5.0/src/crypto.rs:272:59
    |
272 |         let state = to_enc_error!(generichash::State::new(32, key), "Failed to init hash")?;
    |                                                           ^^
    |                                                           |
    |                                                           expected enum `std::option::Option`, found integer
    |                                                           help: try using a variant of the expected enum: `online_managers::_::_serde::__private::Some(32)`
    |
    = note: expected enum `std::option::Option<usize>`
               found type `{integer}`

Some errors have detailed explanations: E0308, E0423, E0608.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `etebase` due to 4 previous errors
```

</details>

**Comment by tasn** [comment]
0.5.1. Will do in a sec.
